xtm/xtm_api.h

```diff
@@ -0,0 +1,209 @@
+#pragma once
+/*
+ * Copyright 2010-2021, Tarantool AUTHORS, please see AUTHORS file.
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the following
+ * conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above
+ *    copyright notice, this list of conditions and the
+ *    following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ *    copyright notice, this list of conditions and the following
+ *    disclaimer in the documentation and/or other materials
+ *    provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY <COPYRIGHT HOLDER> ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * <COPYRIGHT HOLDER> OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+#include <stdlib.h>
+#include <stdbool.h>
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/** \cond public */
+
+/**
+ * Flags, that define library behavior
+ */
+enum {
+	/**
+	 * Flag indicates, that xtm_delete must close all read file
+	 * descriptors, otherwise they must be closed by user before
+	 * calling xtm_delete.
+	 */
+	XTM_QUEUE_NEED_TO_CLOSE_READFD = 1 << 0,
+	/**
+	 * Flag indicates, that producer thread want to receive
+	 * notifications when queue is not full.
+	 */
+	XTM_QUEUE_PRODUCER_NEED_NOTIFICATIONS = 1 << 1,
+};
+
+/**
+ * Opaque struct, that represent unidirectional, single-writer-single-reader
+ * queue implementation with event loop integration ability.
+ */
+struct xtm_queue;
+
+/**
+ * Typedef for often used function pointer, used in push_fun/invoke_funs_all
+ * xtm used pattern.
+ */
+typedef void (*xtm_queue_fun_t)(void*);
+
+/**
+ * Create instance of struct xtm_queue.
+ * @param[in] size - queue size, must be power of two.
+ * @param[in] flags - flags defining library behavior.
+ * @retval    pointer to new xtm_queue or NULL in case of error.
+ */
+struct xtm_queue *
+xtm_queue_new(unsigned size, unsigned flags);
+
+/**
+ * Free queue and close its internal fds.
```

Comment:
The same as I said in the readme - it does not always close them. And since you have docs duplicated in the readme and here, you have the same mistakes in 2 places.

Reply:
Fixed